solver/src/reasoners/cp/mod.rs

```diff
@@ -31,56 +32,68 @@ struct LinearSumLeq {
 
 impl LinearSumLeq {
     fn get_lower_bound(&self, elem: SumElem, domains: &Domains) -> IntCst {
-        debug_assert!(elem.or_zero || domains.present(elem.var) == Some(true));
-        let int_part = match elem.factor.cmp(&0) {
-            Ordering::Less => domains
-                .ub(elem.var)
-                .saturating_mul(elem.factor)
-                .clamp(INT_CST_MIN, INT_CST_MAX),
-            Ordering::Equal => 0,
-            Ordering::Greater => domains
-                .lb(elem.var)
-                .saturating_mul(elem.factor)
-                .clamp(INT_CST_MIN, INT_CST_MAX),
-        };
-        match domains.present(elem.var) {
-            Some(true) => int_part, // note that if there is no default value, the variable is necessarily present
-            Some(false) => 0,
-            None => 0.min(int_part),
+        if let Some(var) = elem.var {
+            debug_assert!(!domains.entails(elem.lit) || domains.present(var) == Some(true));
+            let int_part = match elem.factor.cmp(&0) {
+                Ordering::Less => domains
+                    .ub(var)
+                    .saturating_mul(elem.factor)
+                    .clamp(INT_CST_MIN, INT_CST_MAX),
+                Ordering::Equal => 0,
+                Ordering::Greater => domains
+                    .lb(var)
+                    .saturating_mul(elem.factor)
+                    .clamp(INT_CST_MIN, INT_CST_MAX),
+            };
+            match domains.present(var) {
+                Some(true) => int_part, // note that if there is no default value, the variable is necessarily present
+                Some(false) => 0,
+                None => 0.min(int_part),
+            }
+        } else {
+            // If None, the var value is considered to be 1
+            elem.factor
```

Comment:
What if `elem.lit` is false ? or unset ?

Reply:
I think indeed it is wrong for both. 
I only looked at it quickly, but I believe that 

 - At line 48, you should not branch on the presence literal but on the `elem.lit` literal.
 - you should the same thing in the else branch (before this patch it was done coherently in both cases).

Reply:
The question really is: what is the lowest possible value that this term (`lit * factor *var`) may have?

Reply:
Fixed in 4bf4851